src/strategyHelpers/canary/podCanaryHelper.ts

```diff
@@ -21,49 +21,57 @@ export async function deployPodCanary(
       throw Error('Percentage must be between 0 and 100')
 
    for (const filePath of filePaths) {
-      const fileContents = fs.readFileSync(filePath).toString()
-      const parsedYaml = yaml.safeLoadAll(fileContents)
+      const fileContents = fs.readFileSync(filePath, 'utf8')
+      const parsedYaml = yaml.loadAll(fileContents)
```

Comment:
💡 I am guessing but here `loadAll` is the method currently available which was previously `safeLoadAll` in newer version?

Reply:
it looks like it. we can link this migration guide for clarity https://github.com/nodeca/js-yaml/blob/master/migrate_v3_to_v4.md

Reply:
if loading the yaml is a failable operation we should wrap this with a try catch?

Reply:
yeah safeLoadAll is deprecated in the newer version. I think it makes sense to add try catch!